### PR TITLE
Unify prediction of object detectors

### LINKED ISSUE #655: Enable Consistent Outputs From TF/Pytorch Faster-RCNN's
**Is your feature request related to a problem? Please describe.**
The outputs of ART's [TF](https://github.com/Trusted-AI/adversarial-robustness-toolbox/blob/1.4.1/art/estimators/object_detection/tensorflow_faster_rcnn.py#L371-L380) and [Pytorch](https://github.com/Trusted-AI/adversarial-robustness-toolbox/blob/1.4.1/art/estimators/object_detection/pytorch_faster_rcnn.py#L220-L223) Faster R-CNN's are of a different format. It would be very useful if I could call `.predict()` using either a Keras or Pytorch detector and return an output in a consistent format.

**Describe the solution you'd like**
Ideally the TF Faster R-CNN would (or could, i.e. if an optional kwarg was added) return an output in the form currently returned by the Pytorch Faster R-CNN. To be more specific: a list of dictionaries (one dictionary for each image) each containing the following keys: `boxes`, `scores`, `labels` as described in the lines highlighted in the Pytorch link above.

edit: another aspect of establishing consistency is the format of the predicted boxes (e.g. [xmin, ymin, xmax, ymax] and whether or not the boxes are normalized by image height/width).

### PR DESCRIPTION
**Pull request by beat-buesser**
Signed-off-by: Beat Buesser <beat.buesser@ie.ibm.com>

# Description

Update the return of method `predict` in `art.estimators.object_detection` to adhere to common label format.

This pull request also makes modification to enable the combination of `ProjectedGradientDescent` and `TensorFlowFasterRCNN`.

Fixes #655

## Type of change

Please check all relevant options.

- [x] Improvement (non-breaking)
- [ ] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by lcadalzo — APPROVED** [review]
LGTM. Confirmed that the output formats of the TF and PyTorch Faster R-CNN's match

**Comment by beat-buesser** [comment]
@lcadalzo Thank you for your review!
